lcm/lcm_coretypes.h

```diff
@@ -7,6 +7,12 @@
 #include <stdlib.h>
 
 #ifdef __cplusplus
+#if defined(__clang__)
+// Suppress warnings about C-style casts, since this code needs to build in
+// both C and C++ modes
+#pragma clang diagnostic push
```

Comment:
Please also consider using [`#pragma GCC`](https://clang.llvm.org/docs/UsersManual.html#pragma-gcc-diagnostic) instead (and corresponding change to `#if defined(__clang__)`; I believe `__GNU_C__` can be used), since this should plausibly be done for GCC as well as clang. (GCC is arguably wrong suppressing warnings in `extern "C"` code; using GCC's pragmas, which are also supported by clang, will proactively guard against GCC changing this behavior.)

Reply:
> (GCC is arguably wrong suppressing warnings in extern "C" code; using GCC's pragmas, which are also supported by clang, will proactively guard against GCC changing this behavior.)

@fbudin69500 had pointed out https://gcc.gnu.org/ml/gcc-patches/1999-07n/msg00229.html
I'm not sure if they didn't mean to do it.

> I believe __GNU_C__ can be used

If you mean `__GNUC__` it's defined in both GCC and clang. 
`clang-4.0 -x c /dev/null -dM -E | grep __GNUC__`

Reply:
GCC has implemented this on purpose, as this patch show. I looked in their source code and the implementation is still the same. They are more lenient than Clang on that matter. This does not say that they may not change their implementation at some point, but at least it is not a bug that they are unaware of. As for the bug submission on the Clang bug tracker, the bug has actually been reopened by the reviewer. The issue is up for discussion.  As a side note, Visual Studio does not take a stand in this battle as it does not implement this flag, at least as far as I found.

Reply:
Ah... well, clearly there is room for debate here as to what constitutes "correct" behavior. Anyway, I don't feel strongly about doing it for both compilers, but OTOH it costs basically nothing to support both, since clang supports GCC's pragmas. Absent @ashuang chiming in with an opinion, feel free to make your own decision.

> If you mean `__GNUC__` [...]

Yes, my bad :smile:. (This is what I get for going by memory rather than looking it up...)